### PR TITLE
Clean-up + Refactorings

### PR DESCRIPTION
**Pull request by AHReccese**
#### What does this implement/fix? Explain your changes.
In this PR, I apply internal clean-up and refactoring to better cover corner cases, improve robustness.

- Token validation at construction so callers cannot reach the network with empty/invalid credentials.
- Regression tests covering token validation and CWD preservation.
- upload now builds the template package in an isolated temporary directory and restores CWD + cleans up via finally.
- Credentials are passed to twine via subprocess env= instead of mutating os.environ
- pre-existing TWINE_* are stripped to prevent external overrides.
- First subprocess failure is reported as-is (no longer overwritten by next commands).
- Template generators updated to reflect reserver's real Python support (>=3.7, 3.7–3.14).
- Default package metadata in params.py reworked to stop referencing unrelated third-party URLs/emails.
- Test names shortened and explanations moved to inline comments.

### CODE REVIEW

**Comment by AHReccese** [comment]
Hi Sadra @sadrasabouri, 
I tried to add proper comments to explain the changes in the code, feel free to ask any questions you might have.

Best,
AmirH

File: reserver/params.py [hunk 1]
```diff
@@ -5,19 +5,21 @@
...
     "description": "This name has been reserved using Reserver",
     "author": "Development Team",
-    "author_email": "test@test.com",
-    "url": "https://url.com",
-    "download_url": "https://download_url.com",
-    "source": "https://github.com/source",
+    "author_email": "reserver@openscilab.com",
+    "url": RESERVER_REPO_URL,
+    "download_url": RESERVER_REPO_URL,
+    "source": RESERVER_REPO_URL,
```
Comment (sadrasabouri) [inline reserver/params.py:15]: Is it actually better to have reserved information for the reserved package? I don't think so.

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Looks good overall. I only have a minor suggestion.

File: reserver/params.py (hunk 1 above)
Comment (AHReccese) [reply reserver/params.py:15]: @sadrasabouri 
Thank you for the comment. As noted earlier, these fields must contain valid URLs, emails, etc., otherwise PyPI rejects the package. I also want to avoid unintentionally referencing or promoting external entities. For that reason, I used the `reserver`’s own values, they are valid, internally consistent, and do not attribute anything to third parties. Users can override these fields during their own usage.

I really appreciate your input here, as I’m not entirely certain this is the best approach.

Comment (sadrasabouri) [reply reserver/params.py:15]: Got it. I tink this is the best we can do for now.

**Review by sadrasabouri — APPROVED** [review]
